wallet-connect/src/client.rs

```diff
@@ -221,6 +221,53 @@ impl WCMiddleware<Provider<Client>> {
     pub fn new(client: Client) -> Self {
         WCMiddleware(Provider::new(client))
     }
+    pub fn with_sender(self, address: impl Into<Address>) -> Self {
+        WCMiddleware(self.0.with_sender(address))
+    }
+
+    pub async fn wc_send_transaction(
```

Comment:
@damoncro instead of this, one can perhaps override `send_transaction` in Middleware implementation, so that it's not confusing that there are two same methods

Reply:
The original `send_transaction` in `MiddleWare` has a different return value, 
```
    async fn send_transaction<T: Into<TypedTransaction> + Send + Sync>(
        &self,
        tx: T,
        block: Option<BlockId>,
    ) -> Result<PendingTransaction<'_, Self::Provider>, Self::Error> {
        self.inner().send_transaction(tx, block).await.map_err(FromErr::from)
    }
```
The return value is `PendingTransaction` which does not work with walletconnect RPC endpoint that will directly return the Transaction Hash. See https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendtransaction 

To avoid confusing, create a new one for walletconnect, prefix with `wc_`.

Reply:
oh, you mean just use the same name?

Reply:
Done https://github.com/cronos-labs/play-cpp-sdk/pull/410/commits/342191e385dfac824181456f896ef11e275e8a2f

Reply:
can't you just return `PendingTransaction` / construct it yourself? https://docs.rs/ethers/1.0.2/ethers/providers/struct.PendingTransaction.html#implementations
 `PendingTransaction`  seems to be just a hash (which you get from WalletConnect) + `Provider` (you may just pass self/&self ?)

Reply:
no, i didn't mean just the same name, i meant overriding that trait method, so one can correctly use it with ethers

Reply:
Done https://github.com/cronos-labs/play-cpp-sdk/pull/410/commits/6d2faf87cb6acbb9dacbf85b389f07fc95e48192